pub/lib/dependabot/pub/update_checker.rb

```diff
@@ -42,26 +42,32 @@ def lowest_resolvable_security_fix_version
       end
 
       def lowest_security_fix_version
-        # TODO: Pub lacks a lowest-non-vulnerable version strategy, for now we simply bump to latest resolvable:
-        # https://github.com/dependabot/dependabot-core/issues/5391
-        relevant_version = latest_resolvable_version
-        return unless relevant_version
-
-        # NOTE: in other ecosystems, the native helpers return a list of possible versions, to which we apply
-        # post-filtering. Ideally we move toward a world where we hand the native helper a list of ignored versions
-        # and possibly a flag indicating "use min version rather than max". The pub team is interested in supporting
-        # that. But in the meantime for internal consistency with other dependabot ecosystem implementations I kept
-        # `relevant_versions` as an array.
-        relevant_versions = [relevant_version]
-        relevant_versions = Dependabot::UpdateCheckers::VersionFilters.filter_vulnerable_versions(relevant_versions,
-                                                                                                  security_advisories)
-        relevant_versions.min
+        # Don't attempt to do security updates for git dependencies.
+        return nil if git_revision? dependency.version
+        # If the current version is not vulnerable, we stay on it.
+        return version_unless_ignored dependency.version unless vulnerable?
+
+        e = dependency_services_smallest_update
+        return nil if e.nil?
+
+        upgrade = e.find { |u| u["name"] == dependency.name }
+
+        version = upgrade["version"]
+        # TODO: Should we do this upgrade even if the found version is ignored?
```

Comment:
Ignore conditions also apply to security updates, as per the [docs here](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#ignore):

> Setting this option will also affect pull requests for security updates to the manifest files of this package manager, unless you use `target-branch` to check for version updates on a non-default branch.

(The `target-branch` part is handled outside of core, we only pass in ignore conditions relevant to the branch that's getting updated)

For reference you can also see that we do this for npm [here](https://github.com/dependabot/dependabot-core/blob/main/npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/latest_version_finder.rb#L71C1-L72)

Reply:
Thanks for the info. Will remove the TODO :)